examples/platform_channel/macos/Runner/MainFlutterWindow.swift

```diff
@@ -0,0 +1,19 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Cocoa
+import FlutterMacOS
+
+class MainFlutterWindow: NSWindow {
+  override func awakeFromNib() {
+    let flutterViewController = FlutterViewController.init()
+    let windowFrame = self.frame
+    self.contentViewController = flutterViewController
+    self.setFrame(windowFrame, display: true)
+
+    BatteryLevelPlugin.register(with: flutterViewController.registrar(forPlugin: "BatteryLevel"))
```

Comment:
This is nitpicking (sorry), but I wonder if we should call this something other than `BatteryLevelPlugin`. A user may see this and think "Oh I have to manually register plugins!". Perhaps call this `BatteryLevelService` to avoid plugin confusion? I don't feel strongly about this, feel free to keep as is.

Reply:
Technically, it _is_ a plugin since it implements `FlutterPlugin` and all the usual protocols a plugin does, we just don't get registration for free since it's not a separate package registered in the pubspec, so doesn't get autogenerated into the usual file. 

I've added a comment to clarify this to readers.